supplier/views.py

```diff
@@ -0,0 +1,44 @@
+from django.shortcuts import render
+from supplier.models import Supplier
+from .forms import SupplierForm
+
+
+# def index(request):
+#     form = SupplierForm()
+#     context = {'form':form}
+#     return render(request, 'supplier/supplier_sign_up.html',context)
+# def login_page(request):
+#     return render(request, 'supplier/supplier_sign_up.html')
+
+
+def signup_supplier(request):
+    form = SupplierForm()
+    context = {'form': form, 'message': ['username should be longer than 6',
+                                         'password should be longer than 6',
+                                         'business name should be longer than 6']}
+    if request.method == 'POST':
+        form = SupplierForm(request.POST)
+        user_name = request.POST.get('user_name')
+        password = request.POST.get('password')
+        first_name = request.POST.get('first_name')
+        last_name = request.POST.get('last_name')
+        business_name = request.POST.get('business_name')
+        supplier = Supplier(user_name=user_name,
+                            password=password,
+                            first_name=first_name,
+                            last_name=last_name,
+                            business_name=business_name)
+        if form.is_valid():
+            Supplier.save_supplier(supplier)
+            return render(request, 'buy_together_app/main.html')
+        else:
+            try:
+                Supplier.save_supplier(supplier)
+            except Exception:
+                context['message'].append('User name is taken')
+                return render(request, 'supplier/supplier_sign_up.html', context)
+    return render(request, 'supplier/supplier_sign_up.html', context)
```

Comment:
if sign up was successful should return to home page

Reply:
Its already made look at the is valid.

Reply:
ok, so no need for line 39.

Reply:
There is need for all the code,line 39 is when a username is taken so we update the failure message to the client

Reply:
after line 38 is executed you can "leave" the else, when you leave the else you get to line 40, so you get in both ways to the command `return render(request, 'supplier/supplier_sign_up.html', context)`